### PR TITLE
Add rules_android_ndk

### PR DESCRIPTION
**Pull request by ted-xie**
Adds a BCR entry for rules_android_ndk v0.1.0. sha sum generated with `curl https://github.com/bazelbuild/rules_android_ndk/archive/refs/tags/0.1.0.tar.gz | openssl dgst -sha256 -binary | openssl base64 -A | xargs echo`

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (rules_android_ndk) have been updated in this PR. Please review the changes.

File: modules/rules_android_ndk/metadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "homepage": "https://github.com/bazelbuild/rules_android_ndk",
+  "maintainers": [
+    {
+      "email": "bcr-maintainers@bazel.build",
+      "name": "No Maintainer Specified"
```
Comment (fmeum) [inline modules/rules_android_ndk/metadata.json:6]: Could you add yourself as a maintainer here?

Comment (ted-xie) [reply modules/rules_android_ndk/metadata.json]: Done

**Comment by fmeum** [comment]
@Wyverald CI appears to be completely stuck, any idea what's going on with it?

**Review by fmeum — APPROVED** [review]

**Comment by Wyverald** [comment]
you need a presubmit.yml file (see other modules for examples)

**Comment by ted-xie** [comment]
I've just realized that I've slightly messed up here in two ways:
* I tagged my release as 0.1.0, rather than **v**0.1.0, as appears to be the custom in other repositories
* I did not create a .github/workflows directory in rules_android_ndk, so no release archive was generated, which causes a BCR presubmit failure (since /archive/refs/tags tarballs are unstable)

So it looks like I have to kill the 0.1.0 release, copy the github workflows from the rules-template repo, and properly tag **v**0.1.0.

**Comment by Wyverald** [comment]
I don't think those matter too much (though if you want to fix them, now is the best time).

We could just allow this unstable tarball for this release if you'd like to.

**Comment by bazel-io** [comment]
Hello @Bencodes, @restingbull, @nkoroste, modules you maintain (rules_kotlin) have been updated in this PR. Please review the changes.

**Comment by ted-xie** [comment]
Disregard the rules_kotlin change, I opened the wrong file in my editor.

**Comment by ted-xie** [comment]
The presubmit check is failing like so:

```
/var/lib/buildkite-agent/builds/bk-docker-00b7/bazel-org-repo-root/examples/basic does not exist or is not a directory
```

I'm not sure why this is happening though, since that directory definitely exists in the release tarball:
```
$ tar tzf rules_android_ndk-v0.1.2.tar.gz | grep examples.basic | head -n5
rules_android_ndk-0.1.2/examples/basic/
rules_android_ndk-0.1.2/examples/basic/.bazelrc
rules_android_ndk-0.1.2/examples/basic/BUILD.bazel
rules_android_ndk-0.1.2/examples/basic/MODULE.bazel
rules_android_ndk-0.1.2/examples/basic/MODULE.bazel.lock
```

**Comment by fmeum** [comment]
@ted-xie Could you try to configure a BCR test module instead in presubmit? See https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md#test-module. I'm not sure whether a regular test run can change the working directory.

File: modules/rules_android_ndk/0.1.2/presubmit.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        - "--android_platforms=//:arm64-v8a"
+      build_targets:
+        - "//java/com/app:app"
+    run_test_module_windows:
+      name: "Verify build targets with bzlmod on Windows"
+      platform: "windows"
+      bazel: ${{ bazel }}
+      environment:
+        ANDROID_NDK_HOME: "c:/android_ndk/r25b"
+        BAZELCI_LOCAL_RUN: true
```
Comment (ted-xie) [inline modules/rules_android_ndk/0.1.2/presubmit.yml:32]: This is a workaround for a known issue with remote caching + Windows + Android NDK. We're working on validating a fix internally.

**Comment by ted-xie** [comment]
After much fussing, this PR is finally passing presubmit.

**Review by Wyverald — APPROVED** [review]

**Pull request merged**